calico_kubernetes/calico_kubernetes.py

```diff
@@ -379,17 +379,40 @@ def _get_api_token(self):
         :rtype: str
         """
         logger.info('Getting Kubernetes Authorization')
+
         try:
             with open('/var/lib/kubelet/kubernetes_auth') as f:
                 json_string = f.read()
         except IOError as e:
-            logger.info("Failed to open auth_file (%s), assuming insecure mode" % e)
-            return ""
+            if self._api_root_secure():
+                logger.exception("Failed to open auth_file when API root is "
+                                 "set to secure (%s). Exiting", KUBE_API_ROOT)
+                sys.exit(1)
+            else:
+                logger.warning("No bearer token found")
```

Comment:
I'd combine these two logs into one - there's no need to have them on separate lines.